wgpu-hal/src/dx12/mod.rs

```diff
@@ -137,8 +167,6 @@ impl D3D12Lib {
             <*mut _>::cast(&mut error),
         )
         .ok()
-        // TODO: If there's a HRESULT, error may still be non-null and
-        // contain info.
         .into_device_result("Root signature serialization")?;
```

Comment:
The idea was to check the `error` blob before bailing out with `into_device_result()`, missed this during the previous cleanup. The TODO remains valid.

Reply:
Hmm, is that something we should handle? What would be an example?

The only scenario I can think of is: the blob getting partially filled with errors after which we get an OOM but it's probably ok to error with an OOM instead of reporting the partial errors.

I wish the exact error codes were documented (like in Vulkan), now we have to guess which variants we might run into...

Reply:
I was expecting a similar setup like `Compile()` the error doesn't need to be OOM.

For `Compile()`, I'm not sure which `HRESULT`s to check either. There is:

- The result from `Compile()`;
- Non-null `error` blob;
- Null `blob`;
- Result from `IDxcOperationResult::GetStatus()`.

(and any getters on `IDxcOperationResult`/`IDxcResult` can fail as well)

Reply:
Left a reply in https://github.com/gfx-rs/wgpu/pull/6200#discussion_r1744221644.